### PR TITLE
New BIP: Logarithm of transaction fee limits block size

### PR DESCRIPTION
**Pull request by simondev1**

### CODE REVIEW

**Comment by luke-jr** [comment]
This is NOT BIP 323!

**Comment by luke-jr** [comment]
What is the subject of the bitcoin-dev mailing list discussion of this proposal?

Technical soundness: What prevents miners from providing inputs to coinjoin with for fake fees?

Backward compatibility: Talks about a hardfork if "everything applied", but I don't see that in the specification.

**Comment by simondev1** [comment]
Mailing List: Not yet sent. I just subscribed the mailing list. Will do that the next few days.

Technical soundness: The same thing that prevents that today. (Not exactly sure though what you meant.) The incentive of miners is to fill the block only with transactions of largest fees, same as today, there is no incentive to put 1sat/byte transaction into the chain when the block can be filled with 30sat/byte transactions.

Backward compatibility: Improved Text

**Comment by luke-jr** [comment]
Mailing List: For the future, please see BIP 2 - mailing list discussion comes before opening a PR or asking for a number assignment.

Technical soundness: Today, fees are not used in any consensus-critical checks, other than to allow the miner to claim them and no higher amount. So a miner giving himself a fee is simply a wash. With your proposal, however, miners now can game this new algorithm by giving themselves fees in others' transactions.

**Comment by luke-jr** [comment]
(BTW, it's unclear how this works since there is not currently a block size limit, but only a block weight limit. eg, does your algorithm enforce against weight or size?)

**Comment by simondev1** [comment]
Fees come out of unspent satoshis in transactions. A miner cannot just increase fees of a transaction. Such a changed transaction would not be signed correctly. There is no change regarding this.

**Comment by luke-jr** [comment]
They can with cooperation of the transaction sender(s) using a CoinJoin.

**Comment by simondev1** [comment]
Block weight vs block size: This is a good question. Some developer more familiar with this should decide that.

CoinJoin: This BIP has no change that affects anything other than block weight or size. CoinJoin should not be affected by this BIP.

**Comment by luke-jr** [comment]
But CoinJoins do break the assumptions of this BIP (that the fee amount can be known reliably).

**Comment by simondev1** [comment]
Mailing list subject: "new BIP: Self balancing between excessively low/high fees and block size" See [April thread](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2019-April/thread.html)

CoinJoin: That's not the case. CoinJoin is not affected by this BIP. (We do not need know what CoinJoin user payed what part of the fee. But we know exactly the total fee of the transaction just like any normal transaction.)

**Comment by luke-jr** [comment]
This still has a technical soundness problem (miners fudging fees via CoinJoin) to address before it can be assigned a number, and needs a copyright notice/license.

**Comment by junderw** [comment]
How is it "staking BTC for a larger block?"

It's not staking if they have nothing to lose.

**Comment by simondev1** [comment]
Yes true. Didn't know how to describe it better.

File: bip-XXXX.mediawiki [hunk 1]
```diff
@@ -415,18 +416,21 @@ The first part of the table shows how spam is reduced. The part where total sum
 
 ==Backward compatibility==
 
-Soft fork: If applied AND old hardcoded block size limit is kept.
+Soft fork: If applied AND old hardcoded block size limit is kept. (Note: This BIP will most probably not establish itself as softfork among miners, because their money driven incentive is to create a max sized block instead a smaller one.)
```
Comment (luke-jr) [inline bip-XXXX.mediawiki:423]: 1. Miners don't have a say in softforks.
2. This isn't true.

**Comment by luke-jr** [comment]
> Didn't know how to describe it better.

It's a fundamental flaw in your proposal...

File: bip-XXXX.mediawiki (hunk 1 above)
Comment (simondev1) [reply bip-XXXX.mediawiki:423]: >     1. Miners don't have a say in softforks.
> 
>     2. This isn't true.

Disagree.

**Comment by simondev1** [comment]
> > Didn't know how to describe it better.
> 
> It's a fundamental flaw in your proposal...

Disagree.

**Comment by simondev1** [comment]
I think this proposal still is a good idea.

File: bip-XXXX.mediawiki [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+  Status: Draft
+  Type: Standards Track
+  Created: 2019-03-28
+</pre>
+
+==Abstract==
+
+Logarithm of transaction fee limits block size.
+
+If a sender is willing to pay a large fee, then a transaction may be allowed in a block even if blocksize for lower fee-transactions is full. Yet the blocksize is limited harshly by a formula.
```
Comment (murchandamus) [inline bip-XXXX.mediawiki:17]: The abstract is concise, but does not sufficiently describe either the solution or the technical issue being addressed.

File: bip-XXXX.mediawiki [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+
+==Motivation==
+
+# Keep block space small.
+# Waste less with spam transactions.
+# Auto balance Fees: Increase very low fees, Descrease very high fees.
+# Allow larger size when sender pays a lot.
+# Allow wallets to calculate/display how much average free block space there is for each fee price bucket.
+# Allow senders to have more control about how the fee/priority of their transaction will behave, especially in the case of increased adoption in the future.
```
Comment (murchandamus) [inline bip-XXXX.mediawiki:26]: The motivation falls short in describing what inadequacies of the existing protocol are being addressed by this BIP.

File: bip-XXXX.mediawiki [hunk 4]
```diff
@@ -0,0 +1,449 @@
...
+
+==Specification==
+
+Every transaction has to fit into the following block space:
```
Comment (murchandamus) [inline bip-XXXX.mediawiki:30]: Please describe in detail the syntax and semantics of your proposal.

File: bip-XXXX.mediawiki [hunk 5]
```diff
@@ -0,0 +1,449 @@
...
+    Input variable 'transactionFee':
+      type: integer
+      unit: Satoshi
+    Input variable 'transactionSize':
+      type: integer
+      unit: bytes
```
Comment (murchandamus) [inline bip-XXXX.mediawiki:36]: Transaction size was retired as a limit to blocks with the activation of segwit. Transactions today are limited by weight. Please update your proposal to clarify which properties of transactions are relevant to your proposal.

**Review by murchandamus — CHANGES_REQUESTED** [review]
Hello Simon,

at this time your draft does not meet the requirements for a BIP. In my review I noticed that the document does not fulfill the standards described in BIP2, e.g. some headers in the preamble and relevant sections are missing, and the content falls short of fully communicating the solution and addressed issue.

If you are still working on this, I would recommend substantially expanding the explanation of your idea. Possibly a more comprehensive description to the mailing list could garner interest of potential collaborators.

If this submission is no longer being worked on, or substantial improvements are not to be expected, I would recommending closing this PR, as I do not anticipate the proposal being issued a number or merged in its current state.

**Comment by murchandamus** [comment]
I am closing this pull request, because it has not made progress in over four years a year, and my review has been left unanswered for six months. If you are still working on this, please request reopening or open another pull request.

**Pull request closed without merging**